lineapy/plugins/serializers/mlflow_io.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import sys
+from inspect import getmodule
+from typing import Any, Optional
+
+from lineapy.data.types import MLflowArtifactInfo
+from lineapy.utils.config import options
+from lineapy.utils.logging_config import configure_logging
+
+"""
+Serializer and deserializer for MLflow supported model flavors. Only include 
+annotated modules at this time. 
+
+[x] sklearn
+[x] prophet
+[ ] tensorflow (need update current annotation)
+[ ] keras (need update current annotation)
+[ ] h2o
+[ ] gluon
+[x] xgboost
+[ ] lightgbm
+[ ] catboost
+[ ] spacy
+[ ] fastai
+[x] statsmodels
+
+"""
+
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+mlflow_io = {}
+
+try:
+    import mlflow
```

Comment:
I have a non-blocking question about how this works but we should create a ticket if you think its worth it. 

Right now this try catch runs when lineapy module is initialized. This means if a user installs a library during a session because they realize its missing, they must restart their environment and reload the extension to get this to work.

I wonder if we should try to wrap this logic into a decorator that runs a function, catches any mlflow module missing errors, runs this import block, and retries the function again before erroring and decorate all the functions in this file with it. 

This way if a user installs a missing package and simply reruns the cell it may work and they wont have to restart kernel.

Reply:
You are right here, this won't cover the case that users install mlflow after lineapy is initialized. I wouldn't prioritize supporting this scenario either. If users need to read/write mlflow from lineapy, my guess is they are already mlflow users (at least for now).